File: gfg/april2024/c++/02_minimum_absolute_difference.cpp
// CODE

class Solution {
    public:
    void inorder(Node *root, int &res, int& prev) {
        if(!root) return;
        inorder(root -> left, res, prev);
        if(prev != -1) res = min(res, abs(root -> data - prev));
        prev = root -> data;
        inorder(root -> right, res, prev);
    }
    
    int absolute_diff(Node *root) {
        int res = INT_MAX;
        int prev = -1;
        inorder(root, res, prev);
        return res;
    }
};

File: gfg/2024_4_april/python/02_minimum_absolute_difference.py
# CODE

class Solution:
    def inorder(self, root, res, prev):
        if not root:
            return
        self.inorder(root.left, res, prev)
        if prev[0] != -1:
            res[0] = min(res[0], abs(root.data - prev[0]))
        prev[0] = root.data
        self.inorder(root.right, res, prev)
    
    def absolute_diff(self, root):
        res = [float('inf')]
        prev = [-1]
        self.inorder(root, res, prev)
        return res[0]